### ISSUE TITLE
siji not being recognised in font-config

### ISSUE THREAD

**Issue by frankplow** [gstk/siji#28]
I'm on Arch and after a recent update, siji stopped appearing under fontconfig. Looking online, theres also a post on r/archlinux with a couple people reporting a similar issue in the comments.

`fc-list | grep -i siji`
returns nothing, despite
`ls /usr/share/fonts/misc | grep siji`
returning `siji.pcf`

Running `fc-cache -rv` or `fc-cache -fv` shows no new fonts installed in the directory containing siji.

I have `70-yes-bitmaps.conf` in my /etc/fonts/conf.d and no $XDG_CONFIG_HOME/fontconfig is present. Other .pcf bitmap fonts load and function fine. I've tried gzipping siji, as most other .pcf fonts installed were, but that made no difference.

I've tried installing both using the AUR package and the install.sh script (both to /usr/share/fonts/misc & ~/.local/share/fonts/misc), with no difference between the two. 

Running `fc-query /usr/share/fonts/misc/siji.pcf`
returns `Can't query face 4294967295 of font file /usr/share/fonts/misc/siji.pcf`

**Comment by MassimoGirondi** [comment]
Same problem here on Arch Linux!

**Comment by stark** [comment]
does it show up on xlsfonts ?

```sh
xlsfonts | grep siji
```

**Comment by MassimoGirondi** [comment]
In my case, `xlsfonts | grep siji` gives the following:
```-wuncon-siji-medium-r-normal--0-0-75-75-c-0-iso10646-1
-wuncon-siji-medium-r-normal--0-0-75-75-c-0-iso10646-1
-wuncon-siji-medium-r-normal--0-0-75-75-c-0-iso10646-1
-wuncon-siji-medium-r-normal--10-100-75-75-c-80-iso10646-1
-wuncon-siji-medium-r-normal--10-100-75-75-c-80-iso10646-1
-wuncon-siji-medium-r-normal--10-100-75-75-c-80-iso10646-1
```

BTW, I tried to install it on a friend's Manjaro computer and it worked, so I'm start to thinking that it's not strictly related to the font by itself.

**Comment by frankplow** [comment]
Yeah, it shows on xlsfonts for me as well. I also thought it was strange the problem was specific to siji. It works on all other bitmap fonts as far as I can tell besides stlarch. freetype 2 updated last week to add support for multi-color glyphs, so maybe this broke bitmap glyphs somehow?

**Comment by charbelnicolas** [comment]
I am having the exact same problem...

I'm trying to run polybar with the siji font with no results.

It also shows on xlsfonts

**Comment by stark** [comment]
Looks like it's same as #27 

Remove the file:
```
# rm /etc/fonts/conf.d/70-no-bitmaps.conf
```

Then update your font cache

**Comment by frankplow** [comment]
As I said in my original post, I have 70-yes-bitmaps.conf in my conf.d:
`ls /etc/fonts/conf.d | grep bitmap` returns `70-yes-bitmaps.conf`. I've also tried `70-force-bitmaps.conf`, which made no difference.
Also the fact that all other bitmap fonts I've tried are working besides siji & stlarch makes me think that its not a config problem.

**Comment by stark** [comment]
@frankplow that's weird because i didn't even need to symlink the `70-yes-bitmaps.conf` file for siji to work on my system and it shows up under `fc-list` as it should. how about you remove the symlink and update your font cache as root.

**Comment by frankplow** [comment]
@stark I tried it, again no luck. It worked just like yours without `70-yes-bitmaps.conf` or any other configuration until last week. Think I've tried every combo of `sudo` / `70-yes-bitmaps.conf` / `70-force-bitmaps.conf` / `fc-cache -f` / `fc-cache -r` at this point.
Can anyone verify that it works on a distro besides arch with the newest freetype2? It seems too coincidental they made an update which changes glyphs massively and siji broke at the exact same time.

**Comment by radfordwa** [comment]
Same problem on my end, happened after updating freetype I believe. Notably, when running the view script, it returns this error:

Warning: Cannot convert string "-wuncon-siji-medium-r-normal--10-100-75-75-c-80-iso10646-1" to type FontStruct
xfd:  no font to display

**Comment by radfordwa** [comment]
From https://github.com/lucy/tewi-font/issues/35
"FreeType now uses the `defaultChar' property of PCF fonts to set the glyph for the undefined character at glyph index 0 (as FreeType already does for all other supported font formats). As a consequence, the order of glyphs of a PCF font if accessed with FreeType can be different now compared to previous versions."

The bdf version of the font works. Place it in "~/.local/share/fonts" (or wherever your fonts directory is lcoated), run fc-cache and voila, Siji working again. The declaration for Siji might be different in your configurations, in my polybar configuration I had to change "Wuncon Siji" to just "Siji" to use the bdf variant.

Edit: View script is also working again for me.

**Comment by m-wells** [comment]
@radfordwa 
I'm on Arch and have had the exact same issues with Siji not showing up in fc-list (although it shows in xlsfonts) and your solution is the only way I can actually use Siji.

**Comment by MassimoGirondi** [comment]
@radfordwa  I can confirm that your solution work also for me. Thanks!

**Comment by erebe** [comment]
@radfordwa same here, installing the bdf instead of the pcf is working fine :+1:

**Comment by Bravo555** [comment]
Hello, still having this issue. Tried placing `siji.bdf` in both `~/.fonts` and `~/.local/share/fonts`, `fc-cache` reports `$HOME/.local/share/fonts: caching, new cache contents: 1 fonts, 0 dirs`, but `fc-list | grep -i siji` returns nothing.

**Comment by Bravo555** [comment]
Issue resolved, turns out there was a `/etc/fonts/conf.d/70-no-bitmaps.conf` file in my system. After removal, all works well.

**Comment by spdyn1** [comment]
Hi guys,
Same here: removing `/etc/fonts/conf.d/70-no-bitmaps.conf`did the trick. Polybar shows fine.
Thanks

**Comment by nektro** [comment]
Yes this should be added to the wiki :+1:

**Comment by silentjay** [comment]
Same on ubuntu, had to remove /etc/fonts/conf.d/70-no-bitmaps.conf. Agree with @nektro please can someone add this to the wiki it took hours to figure this out.

**Comment by Thiviyan** [comment]
removing /etc/fonts/conf.d/70-no-bitmaps.conf works in debian. 
issue is : fonts looks ugly for github and facebook in firefox at the same time its fine in chrome.

**Comment by ryanerwin** [comment]
>  issue is : fonts looks ugly for github and facebook in firefox at the same time its fine in chrome.

After removing `/etc/fonts/conf.d/70-no-bitmaps.conf` on Ubuntu 21.04, polybar's example icons work, and no change in the Firefox's font rendering.

I found several mentions of this from 2015, but didn't see recent updates.

**Comment by entnuo** [comment]
If you still can't see the icons after replacing `siji.pcf` with `siji.bdf`, modify the `fonts.dir` file in `$HOME/.local/share/fonts` (or where you've installed it).

There will be a line like:
`siji.pcf -wuncon-siji-medium-r-normal--10-100-75-75-c-80-iso10646-1`

Just change `siji.pcf` with `siji.bdf`, so it'll be like this:
`siji.bdf -wuncon-siji-medium-r-normal--10-100-75-75-c-80-iso10646-1`
